src/core/server/cross_compatibility/cross_compatibility_service.ts

```diff
@@ -0,0 +1,117 @@
+    this.log = coreContext.logger.get('cross-compatibility-service');
+  }
+
+  start({ opensearch, plugins }: StartDeps): CrossCompatibilityServiceStart {
+    this.log.warn('Starting cross compatibility service');
+    return {
+      verifyOpenSearchPluginsState: (pluginName: string) => {
+        this.dashboardsPluginName = pluginName;
+        const pluginOpenSearchDeps = plugins.get(pluginName) || {};
+        return this.verifyOpenSearchPluginsState(opensearch, pluginOpenSearchDeps);
+      },
+    };
+  }
+
+  public async getOpenSearchPlugins(opensearch: OpenSearchServiceStart) {
+    // Makes cat.plugin api call to fetch list of OpenSearch plugins installed on the cluster
+    try {
+      const { body } = await opensearch.client.asInternalUser.cat.plugins<any[]>({
+        format: 'JSON',
+      });
+      return body;
+    } catch (error) {
+      this.log.warn(
+        `Cat API call to OpenSearch to get list of plugins installed on the cluster has failed: ${error}`
+      );
+      return [];
+    }
+  }
+
+  public async checkPluginVersionCompatibility(
+    pluginOpenSearchDeps: CompatibleEnginePluginVersions,
+    opensearchInstalledPlugins: CatPluginsResponse
+  ) {
+    const results: CrossCompatibilityResult[] = [];
+    for (const [pluginName, versionRange] of Object.entries(pluginOpenSearchDeps)) {
+      // add check to see if the Dashboards plugin version is compatible with installed OpenSearch plugin
+      const {
+        isCompatible,
+        installedPluginVersions,
+      } = await this.isVersionCompatibleOSPluginInstalled(
+        opensearchInstalledPlugins,
+        pluginName,
+        versionRange
+      );
+      results.push({
+        pluginName,
+        isCompatible: !isCompatible ? false : true,
+        incompatibilityReason: !isCompatible
+          ? `OpenSearch plugin "${pluginName}" in the version range "${versionRange}" is not installed on the OpenSearch for the OpenSearch Dashboards plugin to function as expected.`
+          : '',
+        installedVersions: installedPluginVersions,
+      });
+
+      if (!isCompatible) {
+        this.log.warn(
+          `OpenSearch plugin "${pluginName}" is not installed on the cluster for the OpenSearch Dashboards plugin "${this.dashboardsPluginName}" to function as expected.`
+        );
+      }
+    }
+    return results;
+  }
+
+  private async verifyOpenSearchPluginsState(
+    opensearch: OpenSearchServiceStart,
+    pluginOpenSearchDeps: CompatibleEnginePluginVersions
+  ): Promise<CrossCompatibilityResult[]> {
+    this.log.info('Checking OpenSearch Plugin version compatibility');
+    // make _cat/plugins?format=json call to the OpenSearch instance
+    const opensearchInstalledPlugins = await this.getOpenSearchPlugins(opensearch);
+    const results = await this.checkPluginVersionCompatibility(
+      pluginOpenSearchDeps,
+      opensearchInstalledPlugins
+    );
+    return results;
+  }
+
+  private async isVersionCompatibleOSPluginInstalled(
```

Comment:
The async signature seems not used in this case, could we remove the `async` and make the call to this method without await?

Reply:
Hi @SuZhou-Joe  Thanks for your review. Yeah I don't think I need async here and in another public function. Removed the unused async usages.